compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -334,6 +334,21 @@ impl<'cx, 'tcx> FallibleTypeFolder<'tcx> for QueryNormalizer<'cx, 'tcx> {
         &mut self,
         constant: mir::ConstantKind<'tcx>,
     ) -> Result<mir::ConstantKind<'tcx>, Self::Error> {
-        constant.try_super_fold_with(self)
+        let constant_kind = match constant {
+            mir::ConstantKind::Ty(c) => {
+                let const_folded = c.try_super_fold_with(self)?;
```

Comment:
this seems wrong, we should still normalize constants here, for now this should probably be
```suggestion
                let const_folded = c.try_fold_with(self)?;
```
but with valtrees we have to avoid the conversion to valtrees here.

If you change this you should be able to avoid the CI issues for now